csv2img-exif

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+##
+# This script reads from the existing Wesnoth copyrights CSV file, and writes it to the CreateDate, ModifyDate, Artist, Copyright, and UserComment Exif tags of the images
+# requires exiftool and cwebp
+##
+
+import argparse
+import contextlib
+import csv
+import os
+from pathlib import Path
+from subprocess import check_output
+from datetime import datetime
+import sys
+import re
+
+##
+# csv file layout:
+# [0] = current file modified date -> CreateDate tag (not totally accurate)
+# [1] = file path, relative to the repository root
+# [2] = license name(s) -> Copyright Exif tag
+# [3] = authorship information -> Artist Exif tag
+# [4] = notes -> UserComment Exif tag
+# [5] = new git commit date (not used) -> ModifyDate (today)
+# [6] = md5 hash (not used, not updated)
+##
+
+##
+# Add new licenses to this list:
+# Avoid things like "GNU GPL v2+;CC BY-SA 4.0", unless you mean to dual license
+# under either GNU GPL v2+ or CC BY-SA 4.0.  GNU GPL v2+ and CC BY-SA 4.0 (e.g.
+# a GNU GPL v2+ file with CC BY-SA 4.0 modifications) isn't legally possible.
+##
+known_licenses = (
+        "CC BY-SA 4.0",
+        "CC0",
+        "GNU GPL v2+",
+        )
+# to check that the date has a four-digit pattern (the year), meaning it really is a date
+dv = re.compile('\d\d\d\d/')
+
+# "-all=" to clear out existing metadata before writing the data from the CSV
+def write_exif(target,date_value,copyright_value,artist_value,usercomment_value):
+    if dv.match(date_value) != None:
+        date_value = datetime.strptime(str(date_value),"%Y/%m/%d").strftime("%Y:%m:%d %H:%M:%S")
+    else:
+        date_value = datetime.today().strftime("%Y:%m:%d %H:%M:%S")
+    return str(check_output(["exiftool", "-overwrite_original", "-all=", "-ModifyDate=now", "-CreateDate=%s" % date_value, "-Copyright=%s" % copyright_value, "-Artist=%s" % artist_value, "-UserComment=%s" % usercomment_value, target]), 'UTF-8')
+
+def fix_webp(target):
+    return str(check_output(["cwebp", "-quiet", "-q", "100", target, "-o", target]), 'UTF-8')
```

Comment:
As `utils/optiwebp.py` this should use `-lossless` and `-exact` and probably all other options used there as well.

https://github.com/wesnoth/wesnoth/blob/e67a91caf22429c56cd2012e324cd8ee212b5481/utils/optiwebp.py#L93

Reply:
why?

Reply:
Doesn't look like this got implemented properly yet. The image data should not be changed, right? So we should use `--lossless` and `--exact`.

Reply:
Actually what is the `cwebp` call even for?

Reply:
> Doesn't look like this got implemented properly yet. The image data should not be changed, right? So we should use --lossless and --exact.

It's properly implemented, aside from the typo in "-mt".  Lossless is implied by "-z" and "-exact" only preserves the invisible garbage.

> Actually what is the cwebp call even for?

Exiftool can't yet add RIFF alpha tags to WEBP, so it produces an Extended WEBP without an Alpha flag.

Reply:
I don't really understand why we need different options here but I suppose it shouldn't make a difference.

It would be good to explain the use of cwebp in a comment so that we can tell once it's not needed anymore.

Reply:
I've added a comment